.github/workflows/release.yml

```diff
@@ -60,18 +74,50 @@ jobs:
       - name: Calculate next version
         run: |
           echo "Latest tag: ${{ env.latest_tag }}"
-          IFS='.' read -r major minor patch <<< "${{ env.latest_tag }}"
-          new_minor=$((minor + 1))
-          new_tag="$major.$new_minor.0"
+          IFS='.' read -r major minor patch <<< "${env.latest_tag#v}"
+
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            if [[ "${{ inputs.version_bump }}" == "patch" ]]; then
+              patch=$((patch + 1))
+            else
+              minor=$((minor + 1))
+              patch=0
+            fi
+          else
+            # Default behavior for scheduled runs
+            minor=$((minor + 1))
+            patch=0
+          fi
+
+          new_tag="v$major.$minor.$patch"
           echo "New tag: $new_tag"
           echo "new_tag=$new_tag" >> $GITHUB_ENV
 
-      # This will trigger a deploy via .github/workflows/cd.yml
-      - name: Push New Tag
+      - name: Create and Push Tag
         run: |
           git config user.name "github-actions[bot]"
           git config user.email "github-actions[bot]@users.noreply.github.com"
           git tag ${{ env.new_tag }}
           git push origin ${{ env.new_tag }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_API_TOKEN }}
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.22.x
+
+      - name: Run goreleaser
```

Comment:
Goreleaser/homebrew now happen sequentially rather than in parallel, because speed is not important and it reduces the chance of issues from one step failing and the other succeeding

Reply:
We'll need to confirm whether goreleaser behaves differently if the workflow itself was not triggered from a tag. Likewise with homebrew.

Reply:
Confirmed that this should be fine